### PR TITLE
Frontend auth components

### PR DESCRIPTION
**Pull request by bill-x**
Creating the components for doing google authentication/authorization

### CODE REVIEW

File: frontend/components/auth/google.jsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import React from 'react';
```
Comment (srekhi) [inline frontend/components/auth/google.jsx]: FYI I tend to import { Component } from 'react' and just extend Component but either is fine

File: frontend/components/auth/google.jsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import { Link, withRouter } from 'react-router-dom';
+import { GoogleLogin } from 'react-google-login';
+import PropTypes from 'prop-types';
+
+
+class GoogleAuth extends React.Component {
+    constructor(props) {
+        super(props);
```
Comment (srekhi) [inline frontend/components/auth/google.jsx]: https://tylermcginnis.com/functional-components-vs-stateless-functional-components-vs-stateless-components/ no need to build a full-fledged component just for your learning - dont need to update this

File: frontend/components/auth/google_container.jsx [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+//import { login, logout, signup, clearErrors } from '../../../actions/session_actions';
+import GoogleAuth from './google';
+
+const mapStateToProps = (state) => ({
+  clientId: state.clientId,
+  scopes: state.scopes
+});
+
+const mapDispatchToProps = (dispatch) => {
+  return null;
```
Comment (srekhi) [inline frontend/components/auth/google_container.jsx]: why are we doing this? just remove the mapDispatchToProps if not needed

Comment (bill-x) [reply frontend/components/auth/google_container.jsx]: Didn't implement this part yet, just stubbing it out but will probably add actions here for after login is successful

File: frontend/reducers/auth_reducer.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import merge from 'lodash/merge';
+
+import {
+    RECEIVE_AUTH_CODE
+} from '../actions/auth_actions';
+
+const defaultState = Object.freeze({
+    clientId: '1046169933414-6mfd4o46adn4lcdmk79mue74h8ca3ou9.apps.googleusercontent.com',
```
Comment (srekhi) [inline frontend/reducers/auth_reducer.js]: what is this default client ID?

Comment (bill-x) [reply frontend/reducers/auth_reducer.js]: Yeah, its the public identifier so that the google knows which app is requesting user data

**Pull request merged**